js/simplicity.js

```diff
@@ -51,35 +51,15 @@ Drupal.behaviors.showMore.attach = function (context, settings) {
   });
 };
 
-Drupal.behaviors.selectOrOther = {};
-Drupal.behaviors.selectOrOther.attach = function(context, settings) {
-  // always show 'other' textfield for donation amount if select-or-other is enabled
-  var $donationComponent = $('.donation-amount', context);
-  var $donationSelect = $('.select-or-other-select', $donationComponent);
-  var $donationOtherRadio = $('[value="select_or_other"]', $donationSelect);
-  var $donationRegularRadios = $('input[type=radio]', $donationSelect).not($donationOtherRadio);
-  var $donationOther = $('.select-or-other-other', $donationComponent);
-  // hide by js so it will show up, when js is disabled
-  // therefor no functionality would be hidden
-  $donationOtherRadio.closest('.form-item').addClass('other-amount');
-
-  if ($donationOther.length > 0) {
-    // IE 7 needs bind on document for change
-    $(document).on('click', '.donation-amount .select-or-other-other', function (e) {
-      var value = e.target.value;
-      // validation should be done in another place
-      // trigger change to get picker to update the state
-      $donationOtherRadio.prop('checked', true).trigger('change');
-      $donationOther.addClass('select-or-other-checked');
-    });
-    $donationRegularRadios.on('click', function (e) {
-      $donationOther.removeClass('select-or-other-checked');
-    });
-    // also add class if the other option has already been selected (e.g. when returning to webform page)
-    if ($donationOtherRadio.prop('checked')) {
-      $donationOther.addClass('select-or-other-checked');
-    }
-  }
+Drupal.behaviors.donationAmount = {};
+Drupal.behaviors.donationAmount.attach = function(context, settings) {
+  var $amountComponents = $('.donation-amount', context);
+  // Mark the other checkbox / radio element for styling.
+  $amountComponents.find('.select-or-other-select input[value="select_or_other"]').closest('.form-item').addClass('other-amount');
+  $amountComponents.find('.select-or-other').bind('select-or-other-update', function (event, data) {
```

Comment:
I think it'd be better to use `.on()`, the use of bind has been discouraged from jQuery 1.7 and deprecated in jQuery 3.

Reply:
Oh, I keep mixing those two up. Will change it …